scipy/signal/signaltools.py

```diff
@@ -501,17 +621,39 @@ def convolve(in1, in2, mode='full'):
         # Convolution is commutative; order doesn't have any effect on output
         volume, kernel = kernel, volume
 
+    if method == 'fft' and not _numeric_arrays([volume, kernel]):
+        raise ValueError('convolve can only be called with method=="fft" when'
+                         ' arrays consist of numeric elements (i.e., '
+                         'int/float/etc)')
+
+    if method == 'auto':
+        method = 'fft' if _fftconv_faster(volume, kernel, mode) else 'direct'
+        method = 'direct' if not _numeric_arrays([volume, kernel]) else method
+
+    if method in ('fft', 'auto') and hasattr(np, "complex256"):
+        method = 'direct' if volume.dtype == 'complex256' else method
+        method = 'direct' if kernel.dtype == 'complex256' else method
```

Comment:
if method=auto and they are complex256, use direct.

if method=fft and they are complex256, call fftconvolve anyway and let it raise the exception it usually does